homeassistant/components/media_player/cast.py

```diff
@@ -54,7 +55,8 @@
 
 def _setup_internal_discovery(hass: HomeAssistantType) -> None:
     """Set up the pychromecast internal discovery."""
-    if hass.data.get(INTERNAL_DISCOVERY_RUNNING_KEY, False):
+    hass.data.setdefault(INTERNAL_DISCOVERY_RUNNING_KEY, threading.Lock())
+    if not hass.data[INTERNAL_DISCOVERY_RUNNING_KEY].acquire():
```

Comment:
I think you should pass `blocking=False` here. Otherwise, this thread will never finish.

Reply:
Oh noes 🤕 I even went through all the documentation and then forgot to include that parameter 😑.